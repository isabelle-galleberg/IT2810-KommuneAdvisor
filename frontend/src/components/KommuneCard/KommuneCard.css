.kommuneCard {
  width: 265px;
  height: 150px;
}

.kommuneCard:hover {
  box-shadow: rgba(100, 100, 111, 0.2) 0px 7px 29px 0px;
  transition: box-shadow 0.2s ease-in-out;
}

.kommuneCardRating {
  display: flex;
  flex-direction: row;
  align-items: center;
  position: absolute;
  right: 16px;
  top: 22px;
}

.averageRating {
  color: darkgray;
  font-size: x-small;
  padding-bottom: 3px;
}

.weaponImg {
  max-width: 60px;
}
