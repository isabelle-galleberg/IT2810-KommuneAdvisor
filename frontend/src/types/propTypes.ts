export interface KommuneCardProps {
  id: string;
  name: string;
  county: string;
  weaponImg: string;
  rating: number;
}
