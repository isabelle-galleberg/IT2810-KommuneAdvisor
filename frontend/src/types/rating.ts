export interface Rating {
    name: string;
    rating: Number;
    title: string;
    description: string;
    timestamp: Date;
    kommuneNumber: string;
  }