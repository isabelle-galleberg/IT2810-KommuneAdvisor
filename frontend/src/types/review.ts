export interface Review {
  _id?: string;
  title: string;
  description: string;
  rating: number;
  name: string;
  timestamp?: string;
}
