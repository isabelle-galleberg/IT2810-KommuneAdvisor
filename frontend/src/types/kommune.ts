import { Rating } from "./rating";

export interface Kommune {
  kommuneNumber: string;
  name: string;
  population: number;
  areaInSquareKm: number;
  mapUrl: string;
  logoUrl: string;
  writtenLanguage: string;
  county: string;
  kommuneRating: Rating[];
}