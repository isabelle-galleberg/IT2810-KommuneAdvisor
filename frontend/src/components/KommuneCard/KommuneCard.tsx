import { Card, Image, Text, Group, Button } from '@mantine/core';
import { Link } from 'react-router-dom';
import { KommuneCardProps } from '../../types/propTypes';
import { Rating } from 'react-simple-star-rating';
import './KommuneCard.css';

export default function KommuneCard({
  id,
  name,
  county,
  weaponImg,
  rating,
}: KommuneCardProps) {
  return (
    <Card
      withBorder
      className='kommuneCard'>
      <Group
        noWrap
        spacing={15}>
        <Image
          src={weaponImg}
          className='weaponImg'
        />
        <div>
          <Text
            mt='xs'
            mb='xs'
            weight='700'
            size='md'>
            {name}
          </Text>
          <Text
            size='sm'
            color='dimmed'>
            üìç{county}
          </Text>
          <Link to={`kommune/${id}`}>
            <Button
              variant='light'
              color='blue'
              mt='xs'
              radius='md'>
              Vis mer
            </Button>
          </Link>
        </div>
        <div className='kommuneCardRating'>
          <Rating
            size={20}
            iconsCount={1}
            readonly
            initialValue={1}></Rating>
          <p className='kommuneCardAverageRating'>
            ({rating != 0 ? rating.toFixed(2) : '-'})
          </p>
        </div>
      </Group>
    </Card>
  );
}
