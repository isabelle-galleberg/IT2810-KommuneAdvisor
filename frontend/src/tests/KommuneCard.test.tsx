import renderer from 'react-test-renderer';
import { KommuneCardProps } from '../types/propTypes';
import KommuneCard from '../components/KommuneCard/KommuneCard';
import { BrowserRouter as Router } from 'react-router-dom';

const data: KommuneCardProps = {
  name: 'Oslo',
  county: 'Oslo',
  rating: 4.5,
  weaponImg:
    'https://upload.wikimedia.org/wikipedia/commons/thumb/d/da/Oslo_komm.svg/1024px-Oslo_komm.svg.png',
};

describe('<KommuneCard />', () => {
  it('should match snapshot with data', () => {
    const elem = renderer
      .create(
        <Router>
          <KommuneCard
            name={data.name}
            county={data.county}
            weaponImg={data.weaponImg}
            rating={data.rating}
          />
        </Router>
      )
      .toJSON();

    expect(elem).toMatchSnapshot();
  });
});
