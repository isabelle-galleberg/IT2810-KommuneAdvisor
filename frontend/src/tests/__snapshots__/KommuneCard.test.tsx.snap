// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<KommuneCard /> should match snapshot with data 1`] = `
<div
  className="mantine-Paper-root mantine-Card-root kommuneCard mantine-rph89u"
>
  <div
    className="mantine-Group-root mantine-1hwpv8j"
  >
    <div
      className="mantine-Image-root weaponImg mantine-15po0m8"
      style={
        Object {
          "height": "auto",
          "width": "100%",
        }
      }
    >
      <figure
        className="mantine-11nhzn5 mantine-Image-figure"
      >
        <div
          className="mantine-qqmv3w mantine-Image-imageWrapper"
        >
          <img
            className="mantine-3y8yz3 mantine-Image-image"
            onError={[Function]}
            onLoad={[Function]}
            src="https://upload.wikimedia.org/wikipedia/commons/thumb/d/da/Oslo_komm.svg/1024px-Oslo_komm.svg.png"
            style={
              Object {
                "height": "auto",
                "objectFit": "cover",
                "width": "100%",
              }
            }
          />
        </div>
      </figure>
    </div>
    <div>
      <div
        className="mantine-Text-root mantine-k0qooo"
      >
        Oslo
      </div>
      <div
        className="mantine-Text-root mantine-rwoji2"
      >
        üìç
        Oslo
      </div>
      <a
        href="/kommune/1"
        onClick={[Function]}
      >
        <button
          className="mantine-UnstyledButton-root mantine-Button-root mantine-1nia2br"
          data-button={true}
          type="button"
        >
          <div
            className="mantine-3xbgk5 mantine-Button-inner"
          >
            <span
              className="mantine-qo1k2 mantine-Button-label"
              style={
                Object {
                  "textTransform": undefined,
                }
              }
            >
              Vis mer
            </span>
          </div>
        </button>
      </a>
    </div>
    <div
      className="kommuneCardRating"
    >
      <span
        className="style-module_starRatingWrap__q-lJC"
        style={
          Object {
            "direction": "ltr",
          }
        }
      >
        <span
          aria-hidden="true"
          className="style-module_simpleStarRating__nWUxf react-simple-star-rating"
          style={
            Object {
              "cursor": "",
            }
          }
        >
          <span
            className="style-module_emptyIcons__Bg-FZ empty-icons"
            style={
              Object {
                "color": "#cccccc",
              }
            }
          >
            <svg
              className="star-svg"
              fill="currentColor"
              height={20}
              stroke="currentColor"
              strokeWidth={0}
              viewBox="0 0 24 24"
              width={20}
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
              />
            </svg>
          </span>
          <span
            className="style-module_fillIcons__6---A filled-icons"
            style={
              Object {
                "color": "#ffbc0b",
                "left": 0,
                "transition": "",
                "width": "100%",
              }
            }
            title="5 out of 1"
          >
            <svg
              className="star-svg"
              fill="currentColor"
              height={20}
              stroke="currentColor"
              strokeWidth={0}
              viewBox="0 0 24 24"
              width={20}
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
              />
            </svg>
          </span>
        </span>
      </span>
      <p
        className="kommuneCardAverageRating"
      >
        (
        4.50
        )
      </p>
    </div>
  </div>
</div>
`;
